## Preparation et nettoyage 🧼

Telechargement du dataset

In [0]:
from pyspark.sql import functions as F

In [0]:
filepath = "s3://full-stack-bigdata-datasets/Big_Data/Project_Steam/steam_game_output.json"
df = spark.read.format('json').load(filepath, multiline=True)
spark.conf.set("spark.sql.legacy.timeParserPolicy", "LEGACY")

Selection des infos utiles:

In [0]:
df= df.select("*", "data.*").drop("data")
df.printSchema()

root
 |-- id: string (nullable = true)
 |-- appid: long (nullable = true)
 |-- categories: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- ccu: long (nullable = true)
 |-- developer: string (nullable = true)
 |-- discount: string (nullable = true)
 |-- genre: string (nullable = true)
 |-- header_image: string (nullable = true)
 |-- initialprice: string (nullable = true)
 |-- languages: string (nullable = true)
 |-- name: string (nullable = true)
 |-- negative: long (nullable = true)
 |-- owners: string (nullable = true)
 |-- platforms: struct (nullable = true)
 |    |-- linux: boolean (nullable = true)
 |    |-- mac: boolean (nullable = true)
 |    |-- windows: boolean (nullable = true)
 |-- positive: long (nullable = true)
 |-- price: string (nullable = true)
 |-- publisher: string (nullable = true)
 |-- release_date: string (nullable = true)
 |-- required_age: string (nullable = true)
 |-- short_description: string (nullable = true)
 |-- tags: struct (nul

Changement du format de la date sous format "yyyy/mm/dd" et /100 du prix

In [0]:
df = df.withColumn('release_date_clean', F.to_date(F.col("release_date"), format="yyyy/MM/dd"))
df = df.withColumn("year", F.year(df['release_date_clean']))
df = df.drop("release_date")
df = df.withColumns({'initialprice':F.col('initialprice')/100,"price":F.col("price")/100})

## 📊Analyse Macro

* Quel éditeur a publié le plus de jeux sur Steam et son prix moyen  ? (TOP 10)

In [0]:
publisher_count = df.withColumn("publisher", F.regexp_replace(F.col("publisher"), ", Inc", " Inc"))\
                    .withColumn("publisher", F.regexp_replace(F.col("publisher"), ", INC", " Inc"))\
                    .withColumn("publisher", F.regexp_replace(F.col("publisher"), ", Ltd", " ltd"))\
                    .withColumn("publisher", F.regexp_replace(F.col("publisher"), ", LTD", " LTD"))\
                    .withColumn("publisher", F.regexp_replace(F.col("publisher"), ", LLC", " LLC"))\
                    .withColumn('publisher', F.explode(F.split("publisher",", ")))\
                    .groupBy('publisher')\
                    .agg(F.count('*').alias('gameCount'),
                         F.mean('price').alias('price_mean'))\
                    .orderBy(F.desc('gameCount'))

publisher_count.limit(9).display()

publisher gameCount price_mean Big Fish Games 422 9.497582938388526 8floor 202 4.994950495049513 SEGA 187 16.39850267379673 Square Enix 151 20.52774834437081 Strategy First 151 6.9243708609271595 THQ Nordic 144 16.184652777777767 Choice of Games 140 5.346428571428578 Sekai Project 136 14.18176470588236 Plug In Digital 136 8.196764705882359

Databricks visualization. Run in Databricks to view.

Big fish a produit 422 jeux avec un prix moyen de 9.49$, suivit par 8floor avec 202 jeux et SEGA avec 187 jeux.

* Quels sont les jeux les mieux notés ?

In [0]:
display(df.select('name','positive').sort(F.col("positive").desc()).limit(10))

name positive Counter-Strike: Global Offensive 5943345 Dota 2 1534895 Grand Theft Auto V 1229265 PUBG: BATTLEGROUNDS 1185361 Terraria 1014711 Tom Clancy's Rainbow Six Siege 942910 Garry's Mod 861240 Team Fortress 2 846407 Rust 732513 Left 4 Dead 2 643836

Databricks visualization. Run in Databricks to view.

Counter Strike, Dota, GTA 5

* Y a-t-il des années avec plus de sorties ? Y a-t-il eu plus ou moins de sorties de jeux pendant le Covid par exemple ?

In [0]:
releases_y = df.groupby('year').count().sort(F.col('count').desc())
display(releases_y)

year count 2021 8805 2020 8287 2018 7663 2022 7451 2019 6949 2017 6006 2016 4176 2015 2566 2014 1550 2013 469 2012 344 2009 309 2010 281 2011 267 null 222 2008 159 2007 98 2006 61 2004 6 2005 6 2001 4 2003 3 1999 3 1997 2 2000 2 1998 1 2002 1

Databricks visualization. Run in Databricks to view.

Depuis 2013, on voit qu'il y'a une tendance a la hausse en terme de publication de jeux sur la platforme steam.
Un premier pic est connu en 2018,puis un autre en 2020-2021.

* Comment sont distribués les prix ? Y a-t-il beaucoup de jeux avec une réduction ?

In [0]:
distrib_prix = df.groupBy('price').count().sort(F.col('count').desc())
display(distrib_prix)
df_price = df.select(['initialprice', 'discount', 'price'])
total_count = df_price.count()
nb_discount = df_price.filter(df_price['discount'] > 0).count()
proportion = (nb_discount / total_count) * 100
discount_df = spark.createDataFrame([("With Discount", nb_discount), ("Without Discount", total_count - nb_discount)], ["Category", "Count"])
display(discount_df)

price count 0.0 7780 4.99 6250 9.99 6126 0.99 5243 1.99 4193 2.99 3639 14.99 2878 19.99 2820 3.99 2530 6.99 1856 5.99 1854 7.99 1580 29.99 865 24.99 862 11.99 822 12.99 796 8.99 753 10.99 430 39.99 428 0.49 303 17.99 286 13.99 276 15.99 230 49.99 190 59.99 187 16.99 185 34.99 155 18.99 141 2.49 99 0.59 97 1.49 73 0.51 51 3.49 45 44.99 44 4.49 41 0.79 40 7.49 40 1.19 40 0.69 38 2.09 34 0.74 34 0.9 32 3.74 29 2.39 27 0.89 25 69.99 25 0.5 25 1.59 25 99.99 24 1.79 23 2.79 22 0.54 21 1.24 21 5.0 20 3.59 20 0.28 20 1.39 19 0.37 19 4.79 18 199.99 17 1.0 16 54.99 15 2.19 14 12.49 14 9.74 13 6.49 13 79.99 13 1.74 12 2.24 12 4.19 11 2.59 11 0.29 11 5.09 10 10.0 10 0.41 10 13.49 9 23.99 9 3.0 9 5.59 9 6.29 9 15.0 9 21.99 9 10.49 9 8.49 8 4.2 8 2.0 8 2.69 8 1.29 8 3.89 8 1.69 7 0.39 7 6.69 7 5.24 7 6.79 7 0.38 7 11.24 7 17.49 7 7.79 7 9.89 7 3.19 6 149.99 6 6.39 6 4.0 6 26.99 6 3.24 6 9.9 6 64.99 6 1.04 6 3.39 6 1.5 6 5.49 6 0.31 6 89.99 5 8.39 5 12.74 5 2.5 5 20.99 5 12.59 5 22.99 5 11.69 5 6.24 5 7.19 5 0.53 5 6.0 5 20.0 5 1.09 5 129.99 5 4.39 4 94.99 4 4.24 4 10.79 4 5.19 4 5.39 4 3.09 4 22.49 4 35.99 4 16.24 4 6.74 4 1.03 4 2.44 4 0.77 4 8.24 4 0.91 4 12.0 4 10.39 4 2.71 4 0.84 3 8.0 3 4.93 3 1.54 3 6.66 3 10.04 3 1.97 3 0.64 3 74.99 3 25.0 3 9.59 3 0.57 3 31.99 3 4.89 3 15.29 3 10.19 3 1.34 3 4.54 3 9.79 3 9.09 3 9.95 2 1.89 2 19.95 2 5.84 2 27.99 2 1.94 2 19.9 2 3.77 2 13.29 2 0.45 2 0.61 2 15.19 2 2.54 2 119.99 2 11.04 2 4.94 2 13.59 2 18.74 2 1.58 2 0.75 2 14.49 2 5.1 2 1.23 2 25.99 2 0.67 2 28.99 2 0.56 2 19.0 2 0.81 2 1.47 2 7.14 2 19.49 2 3.29 2 1.33 2 0.3 2 6.59 2 3.79 2 0.83 2 3.97 2 9.0 2 5.94 2 29.98 2 19.96 1 9.49 1 4.88 1 1.42 1 12.79 1 7.0 1 249.0 1 84.99 1 3.5 1 10.25 1 16.39 1 6.1 1 8.79 1 124.99 1 3.33 1 3.88 1 8.54 1 7.36 1 31.96 1 24.79 1 2.63 1 8.7 1 15.1 1 2.9 1 5.69 1 2.94 1 64.95 1 1.4 1 7.06 1 19.69 1 4.33 1 2.3 1 23.39 1 52.74 1 35.0 1 1.35 1 69.03 1 3.03 1 6.23 1 2.03 1 13.37 1 6.19 1 9.87 1 19.79 1 11.89 1 3.14 1 2.36 1 39.9 1 49.69 1 8.74 1 44.95 1 1.11 1 39.0 1 1.07 1 13.13 1 24.59 1 7.95 1 269.99 1 2.45 1 1.88 1 12.9 1 5.57 1 0.8 1 9.69 1 13.39 1 12.4 1 14.39 1 4.63 1 33.66 1 149.0 1 5.58 1 1.45 1 17.77 1 7.5 1 1.98 1 5.73 1 7.69 1 2.16 1 37.49 1 2.74 1 3.69 1 6.31 1 8.9 1 1.87 1 1.95 1 14.5 1 12.95 1 45.0 1 0.62 1 0.98 1 14.95 1 9.11 1 9.98 1 44.98 1 12.24 1 4.3 1 24.49 1 6.98 1 56.24 1 2.93 1 9.29 1 20.24 1 0.76 1 16.66 1 999.0 1 11.49 1 159.99 1 2.14 1 20.04 1 5.65 1 0.65 1 0.71 1 22.6 1 7.64 1 1.32 1 4.95 1 1.36 1 5.29 1 17.24 1 21.09 1 2.22 1 57.95 1 11.19 1 299.9 1 59.95 1 6.8 1 8.98 1 0.44 1 69.95 1 23.97 1 3.91 1 8.19 1 35.95 1 18.89 1 40.0 1 21.24 1 30.0 1 8.88 1 5.9 1 5.14 1 99.0 1 7.77 1 4.59 1 8.09 1 14.38 1 1.91 1 3.86 1 1.55 1 4.64 1 5.79 1 4.28 1 3.6 1 1.83 1 2.29 1 2.88 1 11.9 1 1.64 1 3.75 1 15.58 1 19.84 1 14.73 1 6.36 1 2.51 1 4.85 1 0.95 1 5.5 1 11.39 1 16.0 1 22.79 1 3.73 1 5.77 1 39.39 1 15.74 1 32.99 1 19.97 1 29.9 1 29.95 1

Databricks visualization. Run in Databricks to view.

Category Count With Discount 2518 Without Discount 53173

Databricks visualization. Run in Databricks to view.

Un peu plus de 4% des jeux sont en promo (2518)

*   Quelles sont les langues les plus représentées ?

In [0]:
df = df.withColumn('languages', F.split(F.col('languages'), ','))
languages=df.select(F.explode('languages').alias('Language'))
display(languages.groupBy('Language').count().sort(F.col('count').desc()).limit(10))

Language count English 54646 German 13996 French 13406 Russian 12839 Spanish - Spain 12224 Simplified Chinese 12213 Japanese 10170 Italian 9297 Portuguese - Brazil 6739 Korean 6575

Databricks visualization. Run in Databricks to view.

Anglais, Allemand, Francais, c'est plus de 50% des langues representés.

## Analyse des genres

* Quels sont les genres les plus représentés ?

In [0]:
genre = df.withColumn('genre_list', F.split(F.trim(df['genre']), ', '))
genre=genre.withColumn('Genre', F.explode(genre['genre_list']))
display(genre.groupBy('Genre').count().sort(F.col('count').desc()).limit(10))

Genre count Indie 39681 Action 23759 Casual 22086 Adventure 21431 Strategy 10895 Simulation 10836 RPG 9534 Early Access 6145 Free to Play 3393 Sports 2666

Databricks visualization. Run in Databricks to view.

Top 3 genres : Indie, action, casual

* Existe-t-il des genres qui ont un meilleur ratio d'avis positifs/négatifs ?

In [0]:
df_exploded_genres = df.select(F.explode(F.split('genre', ',')).alias('genre'), 'positive', 'negative')
df_exploded_genres = df_exploded_genres.withColumn('ratio', F.round(F.col('positive') / F.col('negative')))
df_filtered = df_exploded_genres.filter(F.col('ratio') > 1)
display(df_filtered)

genre positive negative ratio Action 201215 5199 39.0 Action 27 5 5.0 Adventure 27 5 5.0 Indie 27 5 5.0 Adventure 4032 646 6.0 Indie 4032 646 6.0 RPG 4032 646 6.0 Strategy 4032 646 6.0 Action 1575 115 14.0 Indie 1575 115 14.0 Simulation 1575 115 14.0 Strategy 1575 115 14.0 Action 1018 462 2.0 Adventure 1018 462 2.0 Indie 1018 462 2.0 RPG 1018 462 2.0 Adventure 18 6 3.0 Indie 18 6 3.0 RPG 18 6 3.0 Strategy 18 6 3.0 Indie 32 12 3.0 RPG 32 12 3.0 Action 4031 315 13.0 Early Access 4031 315 13.0 Action 309 87 4.0 Adventure 309 87 4.0 Free to Play 309 87 4.0 Indie 309 87 4.0 Action 1212 328 4.0 Early Access 1212 328 4.0 Adventure 55 13 4.0 Indie 55 13 4.0 RPG 55 13 4.0 Action 9 6 2.0 Indie 9 6 2.0 RPG 9 6 2.0 Simulation 9 6 2.0 Early Access 9 6 2.0 Action 9 3 3.0 Casual 9 3 3.0 Indie 9 3 3.0 Massively Multiplayer 9 3 3.0 Education 104 21 5.0 Casual 53 12 4.0 Free to Play 53 12 4.0 Indie 53 12 4.0 Strategy 53 12 4.0 Indie 6 1 6.0 Adventure 107 8 13.0 Indie 107 8 13.0 Indie 6 4 2.0 Racing 6 4 2.0 Simulation 6 4 2.0 Sports 6 4 2.0 Indie 5 1 5.0 Strategy 5 1 5.0 Casual 162 4 41.0 Indie 162 4 41.0 Strategy 162 4 41.0 Action 40 19 2.0 Massively Multiplayer 40 19 2.0 Simulation 40 19 2.0 Action 8 3 3.0 Casual 8 3 3.0 Indie 8 3 3.0 Indie 9 3 3.0 RPG 9 3 3.0 RPG 66 39 2.0 Strategy 66 39 2.0 Early Access 66 39 2.0 Adventure 111 20 6.0 Casual 111 20 6.0 Indie 111 20 6.0 Casual 264 27 10.0 Casual 13 2 7.0 Indie 13 2 7.0 Strategy 13 2 7.0 Indie 30 3 10.0 Simulation 30 3 10.0 Strategy 30 3 10.0 Action 5 3 2.0 Indie 5 3 2.0 Adventure 6 2 3.0 Indie 6 2 3.0 RPG 6 2 3.0 Indie 6 3 2.0 Simulation 6 3 2.0 Action 32 2 16.0 Casual 32 2 16.0 Indie 32 2 16.0 Racing 5 2 3.0 Simulation 5 2 3.0 Action 380 79 5.0 Adventure 380 79 5.0 Indie 380 79 5.0 Action 146 24 6.0 Casual 146 24 6.0 Free to Play 146 24 6.0 Indie 146 24 6.0 Action 29 1 29.0 Casual 29 1 29.0 Indie 29 1 29.0 Casual 46 10 5.0 Indie 46 10 5.0 Casual 6 4 2.0 Indie 6 4 2.0 Action 10 5 2.0 Adventure 10 5 2.0 Casual 10 5 2.0 Indie 10 5 2.0 RPG 10 5 2.0 Action 28 2 14.0 Indie 28 2 14.0 Early Access 28 2 14.0 Indie 80 21 4.0 Simulation 80 21 4.0 Strategy 80 21 4.0 Indie 58 25 2.0 Action 29 4 7.0 Indie 29 4 7.0 Action 38 14 3.0 Adventure 38 14 3.0 Indie 38 14 3.0 Early Access 38 14 3.0 Action 6 3 2.0 Casual 6 3 2.0 Indie 6 3 2.0 Massively Multiplayer 6 3 2.0 Simulation 6 3 2.0 Adventure 10 6 2.0 Indie 10 6 2.0 RPG 10 6 2.0 Casual 22 1 22.0 Action 49 6 8.0 Casual 49 6 8.0 Indie 49 6 8.0 Simulation 49 6 8.0 Casual 19 9 2.0 Casual 338 13 26.0 Free to Play 338 13 26.0 Indie 338 13 26.0 Action 6 2 3.0 Adventure 6 2 3.0 Casual 6 2 3.0 Indie 6 2 3.0 Simulation 6 2 3.0 Casual 60 12 5.0 Indie 60 12 5.0 Action 7 3 2.0 Indie 7 3 2.0 Action 167 11 15.0 Indie 167 11 15.0 Simulation 167 11 15.0 Sports 167 11 15.0 Strategy 167 11 15.0 Early Access 167 11 15.0 Indie 11 3 4.0 Strategy 11 3 4.0 Action 5 1 5.0 Adventure 5 1 5.0 Casual 5 1 5.0 Indie 5 1 5.0 RPG 5 1 5.0 Strategy 5 1 5.0 Indie 14 4 4.0 Adventure 3144 144 22.0 Casual 3144 144 22.0 Indie 3144 144 22.0 Adventure 3 1 3.0 Casual 3 1 3.0 Indie 3 1 3.0 Adventure 44 16 3.0 Casual 44 16 3.0 Indie 44 16 3.0 RPG 3 2 2.0 Action 14 1 14.0 Indie 14 1 14.0 Adventure 23 10 2.0 Casual 23 10 2.0 Indie 23 10 2.0 RPG 23 10 2.0 Simulation 23 10 2.0 Free to Play 74 45 2.0 Indie 74 45 2.0 RPG 74 45 2.0 Simulation 74 45 2.0 Strategy 74 45 2.0 Early Access 74 45 2.0 Free to Play 14 3 5.0 Indie 14 3 5.0 Massively Multiplayer 14 3 5.0 Early Access 14 3 5.0 Casual 37 11 3.0 Indie 37 11 3.0 RPG 37 11 3.0 Simulation 37 11 3.0 Strategy 37 11 3.0 Action 14 4 4.0 RPG 29 1 29.0 Indie 26 13 2.0 Simulation 26 13 2.0 Adventure 745 136 5.0 Indie 745 136 5.0 Simulation 745 136 5.0 Strategy 745 136 5.0 Adventure 86 29 3.0 Indie 86 29 3.0 Action 16 2 8.0 Indie 16 2 8.0 Action 11 2 6.0 Indie 11 2 6.0 Strategy 11 2 6.0 Casual 5 2 3.0 Indie 5 2 3.0 Casual 520 172 3.0 Indie 520 172 3.0 Simulation 520 172 3.0 Adventure 20 9 2.0 Indie 20 9 2.0 RPG 20 9 2.0 Strategy 20 9 2.0 Early Access 20 9 2.0 Adventur

Databricks visualization. Run in Databricks to view.

Le genre ayant le meilleur ratio de notes positifs sur les notes négatifs est le genre Action, puis Indie, puis Adventure.

* Certains éditeurs ont-ils des genres favoris ?

In [0]:
# voyons deja les genres par editeurs :
Genre_edit = genre\
    .groupBy('publisher','Genre')\
        .count().sort(F.col('count').desc())
display(Genre_edit)

publisher Genre count Big Fish Games Casual 418 Big Fish Games Adventure 392 8floor Casual 202 Choice of Games RPG 139 Choice of Games Indie 136 HH-Games Casual 132 Laush Studio Indie 124 Choice of Games Adventure 112 Indie 106 Alawar Entertainment Casual 105 Sekai Project Casual 99 Sokpop Collective Indie 97 Slitherine Ltd. Strategy 96 Alawar Entertainment Adventure 95 Reforged Group Indie 88 Sekai Project Indie 88 Laush Studio Casual 87 Devolver Digital Indie 87 PLAYISM Indie 81 SEGA Action 80 Plug In Digital Indie 79 Hosted Games RPG 79 Piece Of Voxel Indie 79 Hosted Games Indie 79 Kagura Games RPG 78 Boogygames Studios Indie 78 Kagura Games Indie 76 Boogygames Studios Casual 76 SA Industry Indie 76 Alawar Entertainment Indie 76 Square Enix Action 75 Artifex Mundi Casual 73 Artifex Mundi Adventure 72 Boogygames Studios Strategy 71 Square Enix RPG 71 Blender Games Indie 70 Blender Games Casual 70 Action 70 Ubisoft Action 70 tinyBuild Indie 70 HH-Games Indie 69 Devolver Digital Action 69 Hosted Games Adventure 69 Tero Lunkka Indie 68 Boogygames Studios Adventure 67 Daedalic Entertainment Indie 66 Casual 64 Piece Of Voxel Casual 62 RewindApp Indie 59 Ziggurat Action 58 THQ Nordic Action 57 Electronic Arts Action 57 Fulqrum Publishing Strategy 56 Paradox Interactive Strategy 56 Garage Games Indie 56 Ripknot Systems Casual 55 Hede Action 54 RewindApp Casual 54 Hede Indie 54 Forever Entertainment S. A. Indie 54 My Way Games Indie 53 Back To Basics Gaming Indie 53 SA Industry Casual 53 Strategy First Strategy 52 BANDAI NAMCO Entertainment Action 52 Sekai Project Adventure 51 William at Oxford Casual 51 SA Industry Action 51 Ghost_RUS Games Indie 50 Garage Games Casual 50 KEMCO Adventure 50 GrabTheGames Indie 49 Aldorlea Games RPG 49 Devolver Digital Adventure 48 Aldorlea Games Indie 48 HH-Games Strategy 48 tinyBuild Action 48 SA Industry Adventure 48 Sometimes You Indie 47 KEMCO RPG 47 Microids Adventure 46 THQ Nordic RPG 46 Hede Simulation 45 PLAYISM Action 45 PsychoFlux Entertainment Indie 45 Ubisoft Adventure 45 MangaGamer Adventure 45 Daedalic Entertainment Adventure 45 My Way Games Casual 45 Atomic Fabrik, Cristian Manolachi Simulation 44 tinyBuild Adventure 44 Volens Nolens Games Indie 44 Atomic Fabrik, Cristian Manolachi Indie 44 玫瑰工作室 Casual 43 Plug In Digital Action 43 Square Enix Adventure 43 HexWar Games Strategy 42 Forever Entertainment S. A. Casual 42 Slitherine Ltd. Simulation 42 Strategy First Adventure 42 玫瑰工作室 Indie 42 Droid Riot Indie 42 Boogygames Studios Simulation 42 Piece Of Voxel Strategy 41 Piece Of Voxel Simulation 41 8Floor Casual 41 Dnovel Adventure 41 Enoops Indie 41 Plug In Digital Adventure 40 ImperiumGame Indie 40 THQ Nordic Strategy 40 Forever Entertainment S. A. Action 40 Anamik Majumdar Indie 40 Hede Casual 40 玫瑰工作室 Adventure 40 Iceberg Interactive Indie 39 Aldorlea Games Adventure 39 Alawar Entertainment Strategy 39 HH-Games Adventure 39 Trinity Project Indie 39 Ghost_RUS Games Casual 38 HH-Games Action 38 GrabTheGames Adventure 38 Atomic Fabrik, Cristian Manolachi Casual 38 THQ Nordic Adventure 38 Tero Lunkka Casual 37 United Independent Entertainment Simulation 37 Bethesda Softworks Action 37 ESDigital Games Casual 37 My Way Games Adventure 37 Adventure 37 MAGIX Software GmbH Video Production 36 Aldorlea Games Casual 36 Hede Adventure 36 Ripknot Systems Indie 35 Luci Entertainment Indie 35 GameHouse Casual 35 familyplay Casual 35 Anamik Majumdar Casual 35 Dnovel Indie 35 Enoops Adventure 35 KOEI TECMO GAMES CO., LTD. Strategy 35 KEMCO Strategy 34 Trinity Project Action 34 Tero Lunkka Adventure 34 Sokpop Collective Action 34 Headup Indie 34 MumboJumbo Casual 34 Dnovel Action 34 HandMade Games Indie 34 Kagura Games Adventure 34 Aerosoft GmbH Simulation 33 Big Black Bear Indie 33 Nacon Sports 33 Atomic Fabrik, Cristian Manolachi Adventure 33 Qumaron Casual 33 SEGA Adventure 33 Milestone S.r.l. Sports 33 Milestone S.r.l. Racing 33 kovalevviktor Indie 33 Strategy First Indie 33 Piece Of Voxel Advent

Big Fish Games les deux genres les plus prédominants sont "casual" et "adventure"

* Quels sont les genres les plus lucratifs ?

In [0]:
p_genre = genre\
    .groupBy('Genre')\
        .sum('price')\
            .sort(F.col('sum(price)').desc()).limit(10)
display(p_genre)

Genre sum(price) Indie 260630.3599998896 Action 183587.68999997684 Adventure 171581.78999998886 Casual 123835.83000003877 Simulation 98516.54000001294 Strategy 91572.01000000977 RPG 86212.95000000563 Early Access 53757.94999999832 Sports 23855.290000000914 Racing 17716.449999999906

Databricks visualization. Run in Databricks to view.

Le genre le plus lucratife est le style Indie, suivie de celui d'action et Aventure

## Analyse de la plateforme

* La plupart des jeux sont-ils disponibles sur Windows/Mac/Linux ?

In [0]:
# Getting nested field platform value and affecting 1 if True and 0 if False
platforms = df\
    .withColumn("Linux", F.when(F.col('platforms').getField('linux'), 1).otherwise(0))\
    .withColumn("Mac", F.when(F.col('platforms').getField('mac'), 1).otherwise(0))\
    .withColumn("Windows", F.when(F.col('platforms').getField('windows'), 1).otherwise(0))

# Calculating the sum of games on each platform
sum_platforms = platforms\
    .agg(F.sum('Mac').alias('Total Mac'), F.sum('Linux').alias('Total Linux'), F.sum('Windows').alias('Total Windows'))

# Number of rows in df
total_games = df.count()

# Displaying the results with total number of games
display(sum_platforms\
    .withColumn('Total number of games', F.lit(total_games)) # Adding the total number of games to the results
    .withColumn('Percentage Mac', F.round(F.col('Total Mac') / total_games * 100,2))
    .withColumn('Percentage Linux', F.round(F.col('Total Linux') / total_games * 100,2))
    .withColumn('Percentage Windows', F.round(F.col('Total Windows') / total_games * 100,2)))

Total Mac Total Linux Total Windows Total number of games Percentage Mac Percentage Linux Percentage Windows 12770 8458 55676 55691 22.93 15.19 99.97

Databricks visualization. Run in Databricks to view.

99.9% des jeux sont sur Windows